cmd/containerd/command/config.go

```diff
@@ -92,44 +92,39 @@ var configCommand = &cli.Command{
 			},
 		},
 		{
-			Name:  "dump",
-			Usage: "See the output of the final main config with imported in subconfig files",
-			Action: func(cliContext *cli.Context) error {
-				config := defaultConfig()
-				ctx := cliContext.Context
-				if err := srvconfig.LoadConfig(ctx, cliContext.String("config"), config); err != nil && !os.IsNotExist(err) {
-					return err
-				}
-
-				return outputConfig(ctx, config)
-			},
+			Name:   "dump",
+			Usage:  "See the output of the final main config with imported in subconfig files",
+			Action: dumpConfig,
 		},
 		{
 			Name:  "migrate",
 			Usage: "Migrate the current configuration file to the latest version (does not migrate subconfig files)",
-			Action: func(cliContext *cli.Context) error {
-				config := defaultConfig()
-				ctx := cliContext.Context
-				if err := srvconfig.LoadConfig(ctx, cliContext.String("config"), config); err != nil && !os.IsNotExist(err) {
-					return err
-				}
-
-				if config.Version < version.ConfigVersion {
-					plugins := registry.Graph(srvconfig.V2DisabledFilter(config.DisabledPlugins))
-					for _, p := range plugins {
-						if p.ConfigMigration != nil {
-							if err := p.ConfigMigration(ctx, config.Version, config.Plugins); err != nil {
-								return err
-							}
-						}
-					}
-				}
-				return outputConfig(ctx, config)
-			},
+			// TODO(vinayakankugoyal): This should not output fields that were not set in the current configuration.
+			Action: dumpConfig,
```

Comment:
@vinayakankugoyal Are you planning to fix the config migrate issue also in the same PR?

Reply:
Nope I was planning on following up on that in a future PR. That's a more nuanced fix where as this one is more straight forwards.